java/ql/src/semmle/code/xml/MavenPom.qll

```diff
@@ -485,3 +485,54 @@ class MavenRepoJar extends File {
     else getVersion().matches(pom.getVersionString() + "%")
   }
 }
+
+class MavenPlugin extends ProtoPom {
+  MavenPlugin() { this.hasName("plugin") }
+}
+
+class MavenShadePlugin extends MavenPlugin {
+  MavenShadePlugin() {
+    this.getGroup().getValue() = "org.apache.maven.plugins" and
```

Comment:
The `groupId` is optional, see https://stackoverflow.com/a/65533111.

Reply:
Thanks, done